sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -102,6 +105,28 @@ case class ScalaUDF(
     }
   }
 
+  /**
+   * Create the converter which converts the scala data type to the catalyst data type for
+   * the return data type of udf function. We'd use `ExpressionEncoder` to create the
+   * converter for typed ScalaUDF only, since its the only case where we know the  type tag
+   * of the return data type of udf function.
+   * @param dataType return type of function
+   * @return the catalyst converter
+   */
```

Comment:
I'm a little bit hesitant about this. I named it as `catalystConverter` because we've had `scalaConverter`. It's more consistent naming in this way. And the reason we named `scalaConverter` instead of `createToScalaConverter` is that it cause the name collision with `CatalystTypeConverters.createToScalaConverter`.

Reply:
Looks okay as it is.

Reply:
Thanks.